tools/accuracy_checker/accuracy_checker/annotation_converters/criteo_kaggle_dac.py

```diff
@@ -0,0 +1,116 @@
+from ..config import NumberField, StringField, PathField, BoolField
+from .format_converter import BaseFormatConverter
+from .format_converter import ConverterReturn
+
+class CriteoKaggleDACConverter(BaseFormatConverter):
+
+    __provider__ = 'criteo_kaggle_dac'
+    # annotation_types = (OAR3DTilingSegmentationAnnotation, )
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'testing_file': PathField(description="Path to testing file."),
+            "batch": NumberField(optional=True, default=128, description="Model batch"),
+            "subsample_size": NumberField(optional=True, default=0,
+                                           description="Limit total record count to batch * subsample size"),
+            "validation": BoolField(optional=True, default=True,
+                                           description="Allows to use half of dataset for validation purposes"),
+            "separator": StringField(optional=True, default='#',
+                                     description="Separator between input identifier and file identifier"),
+            "preprocessed_dir": PathField(optional=False, is_directory=True, check_exists=True,
+                                          description="Preprocessed dataset location")
+        })
+
+        return parameters
+
+    def configure(self):
+        self.src = self.get_value_from_config('testing_file')
+        self.batch = self.get_value_from_config('batch')
+        self.subsample = self.get_value_from_config('subsample_size')
+        self.validation = self.get_value_from_config('validation')
+        self.separator = self.get_value_from_config('separator')
+        self.preprocessed_dir = self.get_value_from_config('preprocessed_dir')
+
+    def convert(self, check_content=False, **kwargs):
+
+        preprocessed_folder = Path(self.preprocessed_dir)
+        input_folder = preprocessed_folder / "bs{}".format(self.batch) / 'input'
+        mask_folder = preprocessed_folder / "bs{}".format(self.batch) / 'mask'
+
+        _ = [folder.mkdir(parents=True) for folder in [input_folder, ] if not folder.exists()]
+
+        annotations = []
+
+        subfolder = 0
+        filecnt = 0
+
+        data = np.load(self.src)
+        X_int = data['X_int']
+        X_cat = data['X_cat']
+        y = data['y']
+        samples, cat_feat = X_cat.shape
+
+        samples = (samples // self.batch) * self.batch
+        start = 0
+
+        if self.subsample:
+            samples = self.subsample * self.batch if samples > self.subsample * self.batch else samples
+        elif self.validation:
+            start = samples // 2
+
+        for i in range(start, samples - self.batch + 1, self.batch):
+            c_input = input_folder / "{:02d}".format(subfolder)
+
+            _ = [folder.mkdir(parents=True) for folder in [c_input,] if not folder.exists()]
```

Comment:
also it should be a cycle, no need in list comprehension (BTW, usual cycle works faster)

Reply:
and even no need in cycle, because you have always have only one element in [c_input] list :)